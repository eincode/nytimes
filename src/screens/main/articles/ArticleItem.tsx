import React from "react"
import {
  TouchableOpacity,
  StyleSheet,
  Image,
  Text,
  TouchableOpacityProps,
  View,
  ImageBackground,
} from "react-native"
import metrics from "../../../config/metrics"

interface IProps extends TouchableOpacityProps {
  article: Article
}

export default (props: IProps) => (
  <TouchableOpacity style={styles.container} {...props}>
    <ImageBackground
      source={{ uri: props.article.multimedia[0].url }}
      style={styles.image}
      resizeMode={"cover"}
    >
      <Text style={styles.headline}>{props.article.headline.main}</Text>
    </ImageBackground>
    <View style={styles.contentContainer}>
      <Text style={styles.source}>{props.article.source}</Text>
      <Text>{props.article.snippet}</Text>
    </View>
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  container: {
    width: metrics.DEVICE_WIDTH * 0.95,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowRadius: 5,
    shadowOpacity: 1,
    elevation: 5,
    backgroundColor: "white",
    borderRadius: 5,
    padding: 10,
    minHeight: 220,
  },

  image: {
    width: metrics.DEVICE_WIDTH * 0.95,
    height: 200,
    position: "absolute",
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5,
    overflow: "hidden",
  },

  contentContainer: {
    marginTop: 200,
  },

  headline: {
    color: "white",
    position: "absolute",
    bottom: 10,
    left: 10,
    fontSize: 20,
    fontWeight: "bold",
  },

  source: {
    color: "grey",
    fontSize: 12,
    marginBottom: 5,
  },
})
