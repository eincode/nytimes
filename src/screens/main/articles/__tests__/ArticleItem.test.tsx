import React from "react"
import renderer from "react-test-renderer"

import ArticleItem from "../ArticleItem"

it("Sort button component renders correctly", () => {
  const button = renderer.create(<ArticleItem />).toJSON()
  expect(button).toMatchSnapshot()
})
