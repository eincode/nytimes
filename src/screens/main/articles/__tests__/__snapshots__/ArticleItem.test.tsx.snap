// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort button component renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "elevation": 5,
      "marginVertical": 5,
      "minHeight": 220,
      "opacity": 1,
      "padding": 10,
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 5,
      "width": 712.5,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "height": 200,
        "overflow": "hidden",
        "position": "absolute",
        "width": 712.5,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "https://static01.nyt.com/newsgraphics/images/icons/defaultPromoCrop.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 200,
            "width": 712.5,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        Object {
          "bottom": 10,
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "left": 10,
          "position": "absolute",
        }
      }
    >
      article_headline
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 200,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 12,
          "marginBottom": 5,
        }
      }
    >
      article_source
    </Text>
    <Text>
      article_snippet
    </Text>
  </View>
</View>
`;
